json/src/value.rs

```diff
@@ -938,29 +938,33 @@ impl<'a> de::MapVisitor for MapDeserializer<'a> {
         }
     }
 
-    fn missing_field<V>(&mut self, _field: &'static str) -> Result<V, Error>
+    fn missing_field<V>(&mut self, field: &'static str) -> Result<V, Error>
         where V: de::Deserialize,
     {
-        // See if the type can deserialize from a unit.
-        struct UnitDeserializer;
+        use std;
 
-        impl de::Deserializer for UnitDeserializer {
-            type Error = Error;
+        struct MissingFieldDeserializer(&'static str);
 
-            fn deserialize<V>(&mut self, mut visitor: V) -> Result<V::Value, Error>
+        impl de::Deserializer for MissingFieldDeserializer {
+            type Error = de::value::Error;
+
+            fn deserialize<V>(&mut self, _visitor: V) -> std::result::Result<V::Value, Self::Error>
```

Comment:
why is this `use std` + `std::result::Result` necessary, it looks like it worked before? This isn't generated code that might not have access to `Result` directly